Target netstandard1.1

berhir opened the pull request:
I have create a initial version targeting the .net standard 1.1.
Now the library can be used on .NET 4.5+, Mono, .NET Core (also on Linux and Mac), UWP Apps and [more](https://github.com/dotnet/standard/blob/master/docs/versions.md#net-standard-versions).

I had to update some dependencies. Mostly in the test project. The test project still targets .NET 4.5.2.
There are still some things to do to get the CI build running.

I hope you like it and feedback is very welcome!

eoin55 commented on Src/HoneyBear.HalClient.Unit.Tests/HoneyBear.HalClient.Unit.Tests.xproj:
```diff
@@ -0,0 +1,22 @@
+  <Import Project="$(VSToolsPath)\DotNet\Microsoft.DotNet.Props" Condition="'$(VSToolsPath)' != ''" />
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>317104fb-db9b-4ced-9d33-e0ddf67e8fa4</ProjectGuid>
+    <RootNamespace>HoneyBear.HalClient.Unit.TestsCore</RootNamespace>
```
Could you rename the root namespace to `HoneyBear.HalClient.Unit.Tests`, please?  Just to keep things consistent.

eoin55 commented on Src/HoneyBear.HalClient/HoneyBear.HalClient.xproj:
```diff
@@ -0,0 +1,21 @@
+  <Import Project="$(VSToolsPath)\DotNet\Microsoft.DotNet.Props" Condition="'$(VSToolsPath)' != ''" />
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>048aa506-06d4-482d-a42a-45678f63f7a0</ProjectGuid>
+    <RootNamespace>HoneyBear.HalClientCore</RootNamespace>
```
Could you rename the root namespace to `HoneyBear.HalClient`, please?

eoin55 commented on Src/HoneyBear.HalClient/project.json:
```diff
@@ -0,0 +1,36 @@
+    "repository": {
+      "type": "git",
+      "url": "git://github.com/eoin55/HoneyBear.HalClient"
+    }
```
Could you add the following, please?

```
  "buildOptions": {
    "xmlDoc": true
  }
```

This should generate the XML docs, which makes the library easier to consume.

eoin55 commented on appveyor.yml:
```diff
@@ -0,0 +1,22 @@
+  # Don't report back to the mothership
+  DOTNET_CLI_TELEMETRY_OPTOUT: 1
+init:
+- ps: $Env:LABEL = "CI" + $Env:APPVEYOR_BUILD_NUMBER.PadLeft(5, "0")
```
I think I'd prefer to just use the plain old version number.  When we append _CI_ it breaks the AssemblyInfo stamping.

berhir commented:
Hi, I am glad you like it!

I fixed the root namespaces and added the XML documentation.

To get the build running I had to add the appveyor.yml but I am not sure if you want to keep it or if you prefer to use the web interface to configure the build?
I used [this sample](https://github.com/StevenLiekens/dotnet-core-appveyor) to get started.

Regarding the version number the .net and nuget teams strongly recommend to use [SemVer](http://semver.org) for versioning and therefor the dotnet CLI is optimized to use it. There is no out of the box solution to set the version number in the project.json file. It's only possible to use the `--version-suffix` parameter to replace the `*` of the version (e.g. `1.0.0-*` gets `1.0.0-CI123` or `1.0.0-beta1`). There is also an [open issue](https://github.com/dotnet/cli/issues/2676) in the dotnet CLI repo regarding this limitation.
But it should be possible to use a script to replace the version number.

What do you think?

One thing that's missing from the appveyor.yml file is the coveralls.net config. If you want to keep the file you need to add your settings.
